daemon/src/setup_contract.rs

```diff
@@ -678,25 +677,28 @@ pub async fn roll_over(
         .try_into_msg3()
         .context("Failed to read Msg3")?;
 
-    Ok(Dlc {
-        identity: sk,
-        identity_counterparty: dlc.identity_counterparty,
-        revocation: rev_sk,
-        revocation_pk_counterparty: other_punish_params.revocation_pk,
-        publish: publish_sk,
-        publish_pk_counterparty: other_punish_params.publish_pk,
-        maker_address: dlc.maker_address,
-        taker_address: dlc.taker_address,
-        lock: dlc.lock.clone(),
-        commit: (commit_tx, msg1.commit, commit_desc),
-        cets,
-        refund: (refund_tx, msg1.refund),
-        maker_lock_amount,
-        taker_lock_amount,
-        revoked_commit,
-        settlement_event_id: announcement.id,
-        refund_timelock: rollover_params.refund_timelock,
-    })
+    Ok((
+        Dlc {
+            identity: sk,
+            identity_counterparty: dlc.identity_counterparty,
+            revocation: rev_sk,
+            revocation_pk_counterparty: other_punish_params.revocation_pk,
+            publish: publish_sk,
+            publish_pk_counterparty: other_punish_params.publish_pk,
+            maker_address: dlc.maker_address,
+            taker_address: dlc.taker_address,
+            lock: dlc.lock.clone(),
+            commit: (commit_tx, msg1.commit, commit_desc),
+            cets,
+            refund: (refund_tx, msg1.refund),
+            maker_lock_amount,
+            taker_lock_amount,
+            revoked_commit,
+            settlement_event_id: announcement.id,
+            refund_timelock: rollover_params.refund_timelock,
+        },
+        rollover_params.funding_fee.clone(),
```

Comment:
Don't understand why we need to pass this one out here. If we've passed in a value as a parameter, the caller already knows it so we don't need to return it again!

Reply:
How else would you like me to store it into the event? Perhaps I don't see something obvious here?

Reply:
`rollover_params` is a parameter to this function. Meaning the caller has access to this value already because they pass it in. Wherever you use this return value, you should also have access to the parameters that you used so you can pass it into the event there.

Reply:
`rollover_params` has a tight scope and all the fields private, would you  prefer me to break this abstraction and allow fishing out that field?

The way I see this is a smaller trade-off, but apart from a slight awkwardness of passing it back, it at least doesn't leak the abstraction. Happy to change it if you consider it not a worthy trade-off.

Reply:
changed to allow `funding_fee()` accessor on the RolloverParams